File: SiriusJavaClient-Web/src/main/java/org/sirius/client/web/classes/WebPopupList.java
/**
 * 
 */
package org.sirius.client.web.classes;

import java.rmi.RemoteException;

/**
 * @author Myk Kolisnyk
 * 
 */
public class WebPopupList extends WebControl {

    /**
     * @param parent
     * @param locator
     */
    public WebPopupList(final Frame parent, final String locator) {
        super(parent, locator);
    }

    /**
     * @param parent
     * @param parentElement
     * @param locator
     */
    public WebPopupList(final Frame parent, final String parentElement,
            final String locator) {
        super(parent, parentElement, locator);
    }

    public void deselectAll() throws RemoteException {
        this.client().core().deselectAll(parentElement, locator);
    }

    public void deselectByIndex(final int arg3) throws RemoteException {
        this.client().core().deselectByIndex(parentElement, locator, arg3);
    }

    public void deselectByValue(final String arg3) throws RemoteException {
        this.client().core().deselectByValue(parentElement, locator, arg3);
    }

    public void deselectByVisibleText(final String arg3) throws RemoteException {
        this.client().core()
                .deselectByVisibleText(parentElement, locator, arg3);
    }

    public String[] getAllOptions() throws RemoteException {
        return this.client().core().getAllOptions(parentElement, locator);
    }

    public String[] getAllSelectedOptions() throws RemoteException {
        return this.client().core()
                .getAllSelectedOptions(parentElement, locator);
    }

    public String getFirstSelectedOption() throws RemoteException {
        return this.client().core()
                .getFirstSelectedOption(parentElement, locator);
    }

    public boolean isMulti() throws RemoteException {
        return this.client().core().isMultiple(parentElement, locator);
    }

    public void selectByIndex(final int arg3) throws RemoteException {
        this.client().core().selectByIndex(parentElement, locator, arg3);
    }

    public void selectByValue(final String arg3) throws RemoteException {
        this.client().core().selectByValue(parentElement, locator, arg3);
    }

    public void selectByVisibleText(final String arg3) throws RemoteException {
        this.client().core().selectByVisibleText(parentElement, locator, arg3);
    }

}

File: SiriusCSharp.Client/Sirius.CSharp.Client.Web/Classes/WebPopupList.cs
//-----------------------------------------------------------------------
// <copyright file="WebPopupList.cs" company="Sirius">
//     Copyright (c) . All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
namespace Sirius.CSharp.Client.Web.Classes
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    /// <summary>
    /// TODO: Add documentation.
    /// </summary>
    public class WebPopupList : WebControl
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="WebPopupList" /> class.
        /// </summary>
        /// <param name="parent">TODO: Add documentation for parent.</param>
        /// <param name="parentElement">TODO: Add documentation for parent element.</param>
        /// <param name="locator">TODO: Add documentation for locator.</param>
        public WebPopupList(Frame parent, string parentElement, string locator)
            : base(parent, parentElement, locator)
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="WebPopupList" /> class.
        /// </summary>
        /// <param name="parent">TODO: Add documentation for parent.</param>
        /// <param name="locator">TODO: Add documentation for locator.</param>
        public WebPopupList(Frame parent, string locator)
            : base(parent, locator)
        {
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <returns>TODO: Add documentation for return value.</returns>
        public bool IsMulti()
        {
            return this.Client().IsMultiple(this.ParentElement, this.Locator);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        public void DeselectAll()
        {
            this.Client().DeselectAll(this.ParentElement, this.Locator);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <param name="arg3">TODO: Add documentation for argument.</param>
        public void DeselectByIndex(int arg3)
        {
            this.Client().DeselectByIndex(this.ParentElement, this.Locator, arg3);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <param name="arg3">TODO: Add documentation for argument.</param>
        public void DeselectByValue(string arg3)
        {
            this.Client().DeselectByValue(this.ParentElement, this.Locator, arg3);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <param name="arg3">TODO: Add documentation for argument.</param>
        public void DeselectByVisibleText(string arg3)
        {
            this.Client().DeselectByVisibleText(this.ParentElement, this.Locator, arg3);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <returns>TODO: Add documentation for return value.</returns>
        public string[] GetAllOptions()
        {
            return this.Client().GetAllOptions(this.ParentElement, this.Locator);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <param name="arg3">TODO: Add documentation for argument.</param>
        public void SelectByIndex(int arg3)
        {
            this.Client().SelectByIndex(this.ParentElement, this.Locator, arg3);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <param name="arg3">TODO: Add documentation for argument.</param>
        public void SelectByValue(string arg3)
        {
            this.Client().SelectByValue(this.ParentElement, this.Locator, arg3);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <param name="arg3">TODO: Add documentation for argument.</param>
        public void SelectByVisibleText(string arg3)
        {
            this.Client().SelectByVisibleText(this.ParentElement, this.Locator, arg3);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <returns>TODO: Add documentation for return value.</returns>
        public string[] GetAllSelectedOptions()
        {
            return this.Client().GetAllSelectedOptions(this.ParentElement, this.Locator);
        }

        /// <summary>
        /// TODO: Add documentation.
        /// </summary>
        /// <returns>TODO: Add documentation for return value.</returns>
        public string GetFirstSelectedOption()
        {
            return this.Client().GetFirstSelectedOption(this.ParentElement, this.Locator);
        }
    }
}
